src/init/features/project.ts

```diff
@@ -0,0 +1,133 @@
+import * as clc from "cli-color";
+import * as _ from "lodash";
+
+import { FirebaseProject, getProject, listProjects } from "../../firebaseApi";
+import * as logger from "../../logger";
+import { promptOnce, Question } from "../../prompt";
+import * as utils from "../../utils";
+
+const NO_PROJECT = "[don't setup a default project]";
+const NEW_PROJECT = "[create a new project]";
+
+/**
+ * Used in init flows to keep information about the project - basically
+ * a shorter version of FirebaseProject with some additional fields.
+ */
+export interface ProjectInfo {
+  id: string; // maps to projectId
+  label: string;
+  instance: string; // maps to FirebaseProject.resources.realtimeDatabaseInstance
+  location: string; // maps to FirebaseProject.resources.locationId
+}
+
+/**
+ * Get the user's desired project, prompting if necessary.
+ * Returns a ProjectInfo object:
+ *
+ * {
+ *  id: project ID [required]
+ *  label: project display label [optional]
+ *  instance: project database instance [optional]
+ * }
+ */
+async function getProjectInfo(options: any): Promise<ProjectInfo> {
+  // The user passed in a --project flag directly, so no need to
+  // load all projects.
+  if (options.project) {
+    // tslint:disable-next-line:no-shadowed-variable
+    let project: FirebaseProject;
+    try {
+      project = await getProject(options.project);
+    } catch (e) {
+      throw new Error(`Error getting project ${options.project}: ${e}`);
+    }
+    // tslint:disable-next-line:no-shadowed-variable
+    const projectId = project.projectId;
```

Comment:
I really don't like disabling lines over and over again. Since `projectId` and `name` are only used in the object below, can they be inlined?

Reply:
Done